### PR TITLE
Add validation to fixtures based on JSON schema

### PR DESCRIPTION
**Pull request by hannah-bulmer**
Currently supports static schema validation.

Will run chosen fixture against schema and print resulting errors as part of running an action or trigger

### CODE REVIEW

File: spec/test-component/component.json [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-      "main": "./lib/actions/invalidEventEmit.js"
-    },
-    "thrownError": {
-      "main": "./lib/actions/thrownError.js"
-    },
-    "manyErrors": {
-      "main": "./lib/actions/manyErrors.js"
-    },
-    "returnData": {
-      "main": "./lib/actions/returnData.js"
```
Comment (dvd-z) [inline spec/test-component/component.json]: Readd these to component.json pls, they're my tests

File: lib/executables/process.js [hunk 2]
```diff
@@ -3,8 +3,12 @@ const utils = require('../helpers/utils.js');
 const { Emitter, EmitterError } = require('../helpers/emitter.js');
 
 exports.runProcess = async function runProcess(componentPath, fixtureKey, action) {
-  const path = await prompts.getActionPath(componentPath, action);
+  const actionData = await prompts.getActionPath(componentPath, action);
+  const { path } = actionData;
```
[the remark this answers is not in the document]
Comment (hannah-bulmer) [reply lib/executables/process.js]: Because the path variable was already taken in `exec.js`

File: spec/test-component/component.json (hunk 1 above)
Comment (hannah-bulmer) [reply spec/test-component/component.json]: Yours are in `spec-integration`?

File: lib/executables/process.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvd-z) [reply lib/executables/process.js]: I'm not seeing one in `exec.js`? Unless you mean `let func = setup.path`

File: spec/test-component/component.json (hunk 1 above)
Comment (dvd-z) [reply spec/test-component/component.json]: oh I see I didn't notice the earlier move

File: lib/executables/process.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hannah-bulmer) [reply lib/executables/process.js]: You're right. I originally had `const path = require('path')` but must've moved it to the helper file. I'll fix this.

File: README.md [hunk 3]
```diff
@@ -53,14 +53,7 @@ elasticio 1.2.0
...
 
-The only required argument is the `path`, which tells the command where to find the component's action/trigger. This file is expected to export the ``process`` function to be executed. In addition to `process`, the file can also export a `startup`, `init`, and `shutdown` function, which will run the existing ones the order of `startup`, `init`, `process`, and `shutdown`. The returned data from `startup` is accessible in `shutdown`. The function signatures are:
-
-```javascript
-exports.startup = function startup(cfg) { return startupData; };
-exports.init = function init(cfg) { };
-exports.process = function process(msg, cfg, snapshot) { };
-exports.shutdown = function shutdown(cfg, startupData) { };
-```
+The only required argument is the `path`, which tells the command where to find the component's action/trigger. This file is expected to export the ``process`` function to be executed. Fixtures will be initially checked against their respective JSON schema for validation, if it exists.
```
Comment (dvd-z) [inline README.md]: I think you accidentally deleted someone my documentation

Comment (hannah-bulmer) [reply README.md]: Sorry, merge probably messed stuff up :(

File: lib/helpers/utils.js [hunk 4]
```diff
@@ -72,3 +74,53 @@ exports.readComponentJSON = function readComponentJSON(componentPath) {
...
+      return false;
+    }
+    return true;
+  };
+
+  let schema = {};
+
+  const actionOrTrigger = component.actions[actionName] ? 'actions' : 'triggers';
+  const action = component[actionOrTrigger][actionName];
+  if (action.dynamicMetadata) {
```
Comment (jhorbulyk) [inline lib/helpers/utils.js]: Do you validate the `getMetaModel` function at some point?

Comment (hannah-bulmer) [reply lib/helpers/utils.js]: What do you mean? It is validated for existence and signature in `cmp:validate`. Right now, fixtures are only validated against static schemas though. I can open an issue for an enhancement to validate them against the results of getMetaModel in the future.

Comment (jhorbulyk) [reply lib/helpers/utils.js]: I just wanted to make sure that was being done somewhere.

**Pull request merged**